frontend/src/store/Store.ts

```diff
@@ -48,9 +48,15 @@ export interface NamespaceState {
   readonly activeNamespaces: Namespace[];
   readonly filter: string;
   readonly isFetching: boolean;
+  // TODO: Can this be non-optional since we default to empty list?
   readonly items?: Namespace[];
   readonly lastUpdated?: Date;
   readonly namespacesPerCluster: Map<string, string[]>;
+  // The items list filters out namespaces with the same name that exist
+  // on multiple clusters because that is how the namespace selector
+  // currently works but these are actually separate namespaces and
+  // some parts of the UI need the full list unfiltered.
+  readonly unfilteredItems?: Namespace[];
```

Comment:
I'm not sure I love this `unfilteredItems` addition.  Not because you don't need it, but because this whole NamespaceState thing is getting pretty confusing.  Minimally, the fields could use some renaming and comments.  Without studying code how are we supposed to know what 'items' is, or that it's some special filtered list.  Oh, and it's not actually filtered by the `filter` field above :-)  We'd be best off by removing the derived state, like `items` and `namespacesPerCluster`, but the danger is that we may have components off of those state changes.  So, I think we should live with them.  But, what do you think of doing some renaming and commenting here?  Maybe something like:
```

Side note:  While looking at this, I think maybe we have a bug in `EnvoyDetails`.  It seems to be using `items` but I think it may actually want `unfilteredItems`
activeNamespaces  // the namespaces currently selected in NamespaceDropdown
filter  // the filter currently defined in NamespaceDropdown
isFetching
namespaces  // formerly 'unfilteredItems', all namespaces returned from API.getNamespaces
namespaceNames // formerly 'items', Set of unique namespace names, for convenience
lastUpdated
namespacesPerCluster: // map clusterName -> namespaces on cluster
```

Reply:
I'll change where the namespace info comes from so these changes won't be needed anymore.

Reply:
OK, but whatever is still here, please add a few comments to capture the understanding.